packages/light-apps/src/IdentityHeader/IdentityHeader.tsx

```diff
@@ -216,6 +229,32 @@ export function IdentityHeader (props: Props) {
     );
   };
 
+  const isValidUrl = (apiUrl: string): boolean => {
+    return (
+      (apiUrl.length > 5) &&
+      // check that it starts with a valid ws identifier
+      (apiUrl.startsWith('ws://') || apiUrl.startsWith('wss://'))
+    );
+  };
+
+  const onSelectNode = (event: React.SyntheticEvent<HTMLElement, Event>, data: DropdownProps) => {
+    const url = data.value! as string;
+    if (isValidUrl(url)) {
+      const settings = Object.assign(uiSettings.get(), {
+        apiUrl: url
+      });
+      uiSettings.set(settings);
+      setUrl(url);
```

Comment:
We should remove `setUrl`. The flow should be:
- `uiSetting.set(settings)` -> this saves in localStorage
- `window.reload()` -> reloads
- on reload, ContextGate will read from localStorage `settings.apiUrl`

Note:
```
    l.log('setting rpc endpoint to: ', url);
    l.log('new rpc provider is: ', provider);
```
These 2 are actually never logged (actually yes, in the split second before window.location.reload, but that's useless). So we can remove setUrl